### PR TITLE
Feature/tl2023 200

### PR DESCRIPTION
**Pull request by Skumarhpa**

### CODE REVIEW

File: src/Sfa.Tl.ResultsAndCertification.Models/Contracts/AdminDashboard/ReviewChangeRequest.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+﻿using Sfa.Tl.ResultsAndCertification.Common.Enum;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Sfa.Tl.ResultsAndCertification.Models.Contracts.AdminDashboard
+{
+    public  class ReviewChangeRequest
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Models/Contracts/AdminDashboard/ReviewChangeRequest.cs:10]: Why don't we have independent classes for each change types? Are we going to be updating this class each time we include a new change? 

I think we should have a base class with the common fields and then a class for each change type

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs [hunk 2]
```diff
@@ -437,9 +438,24 @@ public async Task<IActionResult> AdminReviewChangesIndustryPlacementAsync(AdminR
             {
                 return View(model);
             }
+            model.LoggedInUser = $"{this.User.FindFirstValue(ClaimTypes.GivenName)} {this.User.FindFirstValue(ClaimTypes.Surname)}";
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs:441]: It cannot be seen here but I noticed in this method that we read from the cache before checking the model is valid. We don't need to call the cache if the model is invalid, so the cache has to be read after the check

File: src/Sfa.Tl.ResultsAndCertification.Application/Services/AdminDashboardService.cs [hunk 3]
```diff
@@ -69,35 +78,89 @@ public async Task<AdminLearnerRecord> GetAdminLearnerRecordAsync(int pathwayId)
...
             if (status > 0)
                 return await _commonService.AddChangelog(CreateChangeLogRequest(request));
             return false;
         }
 
-        private static ChangeLog CreateChangeLogRequest(ReviewChangeStartYearRequest request)
+        public async Task<bool> ProcessChangeIndustryPlacementAsync(ReviewChangeRequest request)
+        {
+            var industryPlacement = await _industryPlacementRepository.GetFirstOrDefaultAsync(p => p.TqRegistrationPathwayId == request.RegistrationPathwayId);
+            request.ChangeType = ChangeType.IndustryPlacement;
```
[the remark this answers is not in the document]
Comment (Skumarhpa) [reply src/Sfa.Tl.ResultsAndCertification.Application/Services/AdminDashboardService.cs]: addressed this comment

File: src/Sfa.Tl.ResultsAndCertification.Models/Contracts/AdminDashboard/ReviewChangeRequest.cs (hunk 1 above)
Comment (Skumarhpa) [reply src/Sfa.Tl.ResultsAndCertification.Models/Contracts/AdminDashboard/ReviewChangeRequest.cs:10]: created independent classes

File: src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs (hunk 2 above)
Comment (Skumarhpa) [reply src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs]: moved this

File: src/Sfa.Tl.ResultsAndCertification.Models/Contracts/AdminDashboard/ReviewChangeRequest.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        public int RegistrationPathwayId { get; set; }
+        public string ContactName { get; set; }
+        public string RequestDate { get; set; }
+        public string ChangeReason { get; set; }
+        public string ZendeskId { get; set; }
+        public string CreatedBy { get; set; }
+        public string Details { get; set; }
+        public ChangeType ChangeType { get; set;}
```
Comment (ibai-gonzalez) [inline src/Sfa.Tl.ResultsAndCertification.Models/Contracts/AdminDashboard/ReviewChangeRequest.cs:19]: Where is this value assigned? (ChangeType property) I can't see it anywhere.
I think it should be inherited by the child classes and return the right value

Comment (Skumarhpa) [reply src/Sfa.Tl.ResultsAndCertification.Models/Contracts/AdminDashboard/ReviewChangeRequest.cs]: updated the code

**Review by ibai-gonzalez — APPROVED** [review]

**Pull request merged**